### PR TITLE
chatterino2: init at unstable-2019-05-11

### PR DESCRIPTION
**Pull request by rexim**
###### Motivation for this change
Chatterino is a chat client for Twitch chat. It aims to be an improved/extended version of the Twitch web chat. It's actively used in the Twitch community.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---
cc @fourtf @pajlada

### CODE REVIEW

File: pkgs/applications/networking/irc/chatterino2/default.nix
Comment (artemist) [inline pkgs/applications/networking/irc/chatterino2/default.nix]: You can use `fetchFromGitHub` instead of `fetchgit` here.

Comment (rexim) [reply pkgs/applications/networking/irc/chatterino2/default.nix]: Oh, ok! After a brief googling I came to a conclusion that `fetchFromGitHub` doesn't support submodules. Just double checked the source code and noticed `fetchSubmodules`. I'll try it now. Thanks!

**Comment by artemist** [comment]
This package otherwise looks good. Thanks for changing that so quickly!

File: pkgs/applications/networking/irc/chatterino2/default.nix
Comment (symphorien) [inline pkgs/applications/networking/irc/chatterino2/default.nix]: please use qtbase etc. as arguments instead of qt5: see https://nixos.org/nixpkgs/manual/#ssec-qt-applications for details.

Comment (rexim) [reply pkgs/applications/networking/irc/chatterino2/default.nix]: @symphorien oh, that is interesting! Thank you for the link!

File: maintainers/maintainer-list.nix
Comment (aanderse) [inline maintainers/maintainer-list.nix]: Please include in a separate commit

File: pkgs/applications/networking/irc/chatterino2/default.nix
Comment (aanderse) [inline pkgs/applications/networking/irc/chatterino2/default.nix]: This is taken care of automatically so you can omit

File: pkgs/applications/networking/irc/chatterino2/default.nix
Comment (aanderse) [inline pkgs/applications/networking/irc/chatterino2/default.nix]: Does the `.pro` file included not take care of installing the binary automatically? If not I'd suggest that might be worth a PR upstream...

File: pkgs/applications/networking/irc/chatterino2/default.nix
Comment (aanderse) [inline pkgs/applications/networking/irc/chatterino2/default.nix]: There was no mention of  64 bit only support in the README and reviewing the `.pro` makes me think you could use `platforms.unix` here instead.

**Review by aanderse — CHANGES_REQUESTED** [review]
Thanks for choosing to contribute to nixpkgs! I have included a few suggestions which I hope will be helpful.

After you're ready with this PR you should rewrite the git history to be 2 commits, the first of which is you adding yourself to the maintainer list and the second adding the package (ie. everything else squashed).

Please let me know if you have any questions.

**Comment by pajlada** [comment]
is "irc" the appropriate category for this package?
The technology that drives the client is IRC, yes, but it's a chat client that can right now only join twitch.tv chats

File: pkgs/applications/networking/irc/chatterino2/default.nix
Comment (pajlada) [reply pkgs/applications/networking/irc/chatterino2/default.nix]: Fixed upstream in the relevant branch, `make install` now installs all relevant stuff to /usr/local (modifiable with `qmake PREFIX=/usr`)

**Comment by rexim** [comment]
@pajlada yeah, I agree. Maybe [instant-messengers](https://github.com/NixOS/nixpkgs/tree/master/pkgs/applications/networking/instant-messengers) is a better category?

**Comment by rexim** [comment]
@aanderse thank you for the remarks! No questions so far. My NixOS machine recently broke, I'll address the remarks as soon as I get it back working. Within a week or so.

**Comment by pajlada** [comment]
@rexim I'm not the most familiar with the categories of packages in NixOS, but from what I can see chatterino fits better under instant-messengers yeah

**Comment by rexim** [comment]
@aanderse I hope I addressed most of the concerns. However there is one more thing. I recenly learned that the version in tag `2.0.4` is very old and the most relevant changes are in the `nightly` branch. So the application has kind of a rolling release nature. I changed the `version` to `2.0.4-nightly` and rev to the latest in `nightly` and will periodically bump it when enough changes in `nightly` are accumulated. Is that an appropriate way of maintaining packages in nixpkgs?

**Comment by aanderse** [comment]
@rexim Awesome work! As for the version, not a problem at all, very common. From the nixpkgs manual:
> If a package is not a release but a commit from a repository, then the version part of the name must be the date of that (fetched) commit. The date must be in "YYYY-MM-DD" format. Also append "unstable" to the name - e.g., "pkgname-unstable-2014-09-23".

**Comment by rexim** [comment]
@aanderse thank you so much for the quick response! Updated the version accordingly.

**Comment by aanderse** [comment]
@GrahamcOfBorg build chatterino2

**Review by aanderse — APPROVED** [review]

**Comment by aanderse** [comment]
@rexim made 2 tiny changes before merging: consistent spacing in `longDescription` and removed redundant `stdenv.lib` text.

Thanks for your contribution to nixpkgs! :tada:

**Pull request merged**
